.github/workflows/build-and-test.yml

```diff
@@ -0,0 +1,219 @@
+name: Code checks
+
+on:
+  pull_request:
+    branches:
+      - "main"
+      - "release-*"
+    paths-ignore:
+      - 'docs/**'
+      - '**.md'
+  workflow_dispatch:
+
+jobs:
+  setup-environment:
+    timeout-minutes: 30
+    runs-on: ubuntu-latest
+    if: ${{ github.actor != 'dependabot[bot]' }}
+    steps:
+      - name: Checkout Repo
+        uses: actions/checkout@v3
+      - name: Setup Go
+        uses: actions/setup-go@v3
+        with:
+          go-version-file: "go.mod"
+      - name: Cache Go
+        id: go-cache
+        uses: actions/cache@v3
+        with:
+          path: |
+            ~/go/bin
+            ~/go/pkg/mod
+          key: go-cache-${{ runner.os }}-${{ hashFiles('**/go.sum') }}
+      - name: Install dependencies
+        if: steps.go-cache.outputs.cache-hit != 'true'
+        run: make -j2 gomoddownload
+      - name: Install Tools
+        if: steps.go-cache.outputs.cache-hit != 'true'
+        run: make install-tools
+  lint-matrix:
+    strategy:
+      matrix:
+        group:
+          - receiver
+#          - processor
+#          - exporter
+#          - connector
+#          - extension
+#          - internal
+#           - other
+    runs-on: ubuntu-latest
+    needs: [setup-environment]
+    steps:
+      - name: Checkout Repo
+        uses: actions/checkout@v3
+      - name: Setup Go
+        uses: actions/setup-go@v3
+        with:
+          go-version-file: "go.mod"
+      - name: Cache Go
```

Comment:
Would it make sense to have it similar to the existing github action we have: https://github.com/kyma-project/telemetry-manager/blob/main/.github/workflows/pr-code-checks.yml#L33 so that we its code has similarity and easy to maintain